src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.CpuUtilizationReader.Windows.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    internal partial class ClrThreadPool
+    {
+        private class CpuUtilizationReader
+        {
+            private struct ProcessCpuInformation
+            {
+                public ulong idleTime;
+                public ulong kernelTime;
+                public ulong userTime;
+            }
+
+            private ProcessCpuInformation _cpuInfo = new ProcessCpuInformation();
+
+            public CpuUtilizationReader()
+            {
+                GetCpuUtilization(); // Call once to initialize the usage buffer
```

Comment:
Don't think this is necessary anymore for the reason indicated. It would be better to initialize this at the top of GateThreadStart before the loop, such that every calculated CPU utilization is a value taken over a 500 ms interval rather than over an arbitrary interval.